pkg/plugin/external/types.go

```diff
@@ -35,6 +35,9 @@ type PluginRequest struct {
 	// Universe represents the modified file contents that gets updated over a series of plugin runs
 	// across the plugin chain. Initially, it starts out as empty.
 	Universe map[string]string `json:"universe"`
+
+	// Config stores the project configuration file.
+	Config string `json:"config"`
```

Comment:
I'm curious, why a string type? Do we have any other types we can use here to make it more explicit what the allowed content is?

Reply:
@camilamacedo86 Fix me if I'm missing something or understanding wrong:

1. I'm wondering if  a literal string of the config content can help make it more general when parsing the data structure from different programs.

2. I'm also wondering how we'd like to define a data structure for config? 
    Shall we directly import the definition from [v3/config](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.11.1/pkg/config/v3/config.go#L55C1-L75C1)? 
    One question here is what exactly content/message a general external plugin will use from the config? 
    How much do we allow the external plugin to modify the config?

Reply:
The biggest concern I have with a string literal of JSON is how do we perform validation on it? How do we ensure that the plugin isn't attempting to use configuration fields that don't exist or they have typos?

Reply:
I'm wondering if we can have a number of clear criteria to help move on? More specifically, I'm curious if we can answer the following questions before implementation:

- What exactly data structure do we want to transfer for config? (Does it have to be a go struct? If yes, then what is the relationship between such with private `cfg` in config v2/v3?)
- Is it necessary to define new methods for `Config` interface so as to help generate the data that might be transferred with the ep?
- How does it affect PluginRequest and PluginResponse?
- What are the boundaries for an ep to touch with config? (Is the config read-only? OR what exactly fields an ep can and cannot touch with?)
- Is it necessary to ALWAYS pass the config to an EP? (Do we need to introduce a mechanism to make it optional?)

Reply:
@Eileen-Yu 

Here, I think, should be config.Config